lib/ansible/plugins/inventory/docker_machine.py

```diff
@@ -0,0 +1,159 @@
+            default: yes
+        split_tags:
+          description: for keyed_groups add two variables as if the tag were actually a key value pair separated by a colon, instead of just a single value.
+          required: False
+          type: bool
+          default: no
+        split_separator:
+          description: for keyed_groups when splitting tags this is the separator to split the tag value on.
+          required: False
+          type: str
+          default: ":"
+'''
+
+EXAMPLES = '''
+# Minimal example
+plugin: docker_machine
+
+# Example using constructed features to create groups
+# keyed_groups may be used to create custom groups
+strict: False
+keyed_groups:
+  - prefix: tag
+    key: 'dm_tags'
+
+# Example using tag splitting where the tag is like 'dm_tag_gantry_component:routinator'
+strict: False
+split_tags: True
+split_separator: ":"
+keyed_groups:
+  - prefix: gantry_component
+    key: 'dm_tag_gantry_component'
+'''
+
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_native
+from ansible.module_utils._text import to_text
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+
+import json
+import re
+import subprocess
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+    ''' Host inventory parser for ansible using Docker machine as source. '''
+
+    NAME = 'docker_machine'
+
+    def _run_command(self, *args):
+        return subprocess.check_output(["docker-machine"] + list(args)).decode('utf-8').strip()
+
+    def _populate(self):
+        self.inventory.add_group('all')
+
+        try:
+            self.nodes = self._run_command('ls', '-q').splitlines()
+            for self.node in self.nodes:
+                self.node_attrs = json.loads(self._run_command('inspect', self.node))
+
+                id = self.node_attrs['Driver']['MachineName']
+                self.inventory.add_host(id)
+                self.inventory.add_group(self.node_attrs['DriverName'])
+                self.inventory.add_host(id, group=self.node_attrs['DriverName'])
+                self.inventory.add_host(id, group='all')
+
+                # Find out more about the following variables at: https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
+                self.inventory.set_variable(id, 'ansible_host', self.node_attrs['Driver']['IPAddress'])
+                self.inventory.set_variable(id, 'ansible_port', self.node_attrs['Driver']['SSHPort'])
+                self.inventory.set_variable(id, 'ansible_user', self.node_attrs['Driver']['SSHUser'])
+                self.inventory.set_variable(id, 'ansible_ssh_common_args', '-o StrictHostKeyChecking=no')
+                self.inventory.set_variable(id, 'ansible_ssh_private_key_file', self.node_attrs['Driver']['SSHKeyPath'])
+
+                # pass '--shell=bash' to workaround 'Error: Unknown shell
+                env_out = self._run_command('env', '--shell=bash', id)
```

Comment:
Isn't `--shell=sh` better, since the chances that `sh` is available are higher than that `bash` is available?

Reply:
I don't think it matters as no shell is actually contacted AFAIK, rather it just determines which format DM uses for the env var assignment text that it generates. I'm fine with `sh` so will adopt your suggestion.

Reply:
Ah, ok. As I said, I've never used docker-machine, so I assumed that it actually connects to the machine (using that shell) and exports the environment from there. If that's just the format, then yes, it really doesn't matter (as long as it is a format you can parse :) ). Both `bash` and `sh` are fine for me, use whatever you want then.

Reply:
According to the [docs](https://docs.docker.com/machine/reference/env/) _"docker-machine env machinename prints out export commands which can be run in a subshell"_, i.e. it doesn't do anything to the actual environment. However, I think `sh` is a as you say a "better" example to use.